flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -461,16 +462,16 @@ AbstractInvokable getInvokable() {
 		return invokable;
 	}
 
-	public StackTraceElement[] getStackTraceOfExecutingThread() {
-		final AbstractInvokable invokable = this.invokable;
-
-		if (invokable == null) {
-			return new StackTraceElement[0];
+	@Override
+	public boolean isBackPressured() {
+		if (invokable == null || consumableNotifyingPartitionWriters.length == 0) {
+			return true;
 		}
-
-		return invokable.getExecutingThread()
-			.orElse(executingThread)
-			.getStackTrace();
+		final CompletableFuture<?>[] outputFutures = new CompletableFuture[consumableNotifyingPartitionWriters.length];
+		for (int i = 0; i < outputFutures.length; ++i) {
+			outputFutures[i] = consumableNotifyingPartitionWriters[i].isAvailable();
+		}
+		return CompletableFuture.allOf(outputFutures).isDone();
```

Comment:
The return value is reverse. If all the outputs are available, then it should return false. The return value `true` means back pressured.